lib/marin/src/marin/export/levanter_checkpoint.py

```diff
@@ -0,0 +1,194 @@
+    VersionedValue,
+    ensure_versioned,
+    this_output_path,
+)
+from marin.resources import CpuOnlyConfig, ResourceConfig, TpuPodConfig
+from marin.training.training import _add_default_env_variables, _add_run_env_variables
+
+logger = logging.getLogger(__name__)
+
+
+@dataclass(frozen=True)
+class ConvertCheckpointStepConfig:
+    """
+    Configuration for converting a single Levanter checkpoint into HuggingFace format.
+    """
+
+    checkpoint_path: str | InputName | VersionedValue[str]
+    trainer: TrainerConfig
+    model: LmConfig
+    resources: ResourceConfig = dataclasses.field(default_factory=CpuOnlyConfig)
+    output_path: str = dataclasses.field(default_factory=this_output_path)  # type: ignore[arg-type]
+    upload_to_hf: bool | str | RepoRef = False
+    tokenizer: str | None = None
+    override_vocab_size: int | None = None
+    config_overrides: dict[str, Any] | None = None
+    save_tokenizer: bool = True
+    use_cpu: bool = False
+    discover_latest: bool = False
+
+
+def convert_checkpoint_to_hf(config: ConvertCheckpointStepConfig) -> None:
+    """
+    Executor entry point that shells into Levanter's export_lm_to_hf utility.
+    """
+
+    default_launch_config = levanter.infra.cli_helpers.load_config()
+
+    checkpoint_path = config.checkpoint_path  # type: ignore[assignment]
+    if config.discover_latest:
+        discovered = discover_latest_checkpoint(checkpoint_path)
+        if not discovered:
+            raise FileNotFoundError(f"Could not discover checkpoint under '{checkpoint_path}'.")
+        checkpoint_path = discovered
+
+    use_cpu = config.use_cpu or isinstance(config.resources, CpuOnlyConfig)
+
+    convert_config = ConvertLmConfig(
+        trainer=config.trainer,
+        checkpoint_path=checkpoint_path,  # type: ignore[arg-type]
+        output_dir=config.output_path,
+        upload_to_hf=config.upload_to_hf,
+        model=config.model,
+        save_tokenizer=config.save_tokenizer,
+        tokenizer=config.tokenizer,
+        override_vocab_size=config.override_vocab_size,
+        config_overrides=config.config_overrides,
+        use_cpu=use_cpu,
+    )
+
+    env = _add_default_env_variables(
+        config.resources.runtime_env.get("env_vars", {}),
+        default_launch_config.env_for_accel(config.resources.accelerator_descriptor() or ""),
+    )
+    env = _add_run_env_variables(env)
+
+    if "JAX_COMPILATION_CACHE_DIR" not in env:
+        marin_prefix = os.environ.get("MARIN_PREFIX")
+        if marin_prefix:
+            env["JAX_COMPILATION_CACHE_DIR"] = os.path.join(marin_prefix, "compilation-cache")
+            logger.info(f"JAX compilation cache enabled at: {env['JAX_COMPILATION_CACHE_DIR']}")
+        else:
+            logger.warning("MARIN_PREFIX environment variable not set. JAX compilation cache will not be configured.")
+
+    hw_config = config.resources.with_env_vars(env)
+
+    @ray.remote(**hw_config.as_remote_kwargs(), max_calls=1)
+    def convert_task():
```

Comment:
Does the export need a TPU slice to run? I guess we don't want to run it on a Ray controller because it's going to try to load everything at once?

Reply:
It's a bit of a weird one:

- Doesn't need the TPU
- Doesn't fit onto the RAM of the head node CPU
- If it runs on a TPU node, it needs to run on a node that doesn't already have a job running because Jax tries to grab the TPU lock even if you pass the env var with devices to only show CPU.